src/chains/ethereum/ethereum/src/transaction-pool.ts

```diff
@@ -300,23 +301,9 @@ export default class TransactionPool extends Emittery.Typed<{}, "drain"> {
     if (secretKey) {
       transaction.signAndHash(secretKey.toBuffer());
     } else if (transaction.v == null) {
-      // if we don't have the secret key and we aren't already signed,
-      // then we are a "fake transaction", so we sign it with a fake key.
-      const from = transaction.from.toBuffer();
-
-      let fakePrivateKey: Buffer;
-      if (from.equals(ACCOUNT_ZERO)) {
-        // allow signing with the 0x0 address...
-        // always sign with the same fake key, a 31 `0`s followed by a single
-        // `1`. The key is arbitrary. It just must not be all `0`s and must be
-        // deterministic.
-        // see: https://github.com/ethereumjs/ethereumjs-monorepo/issues/829#issue-674385636
-        fakePrivateKey = Buffer.allocUnsafe(32).fill(0, 0, 31);
-        fakePrivateKey[31] = 1;
-      } else {
-        fakePrivateKey = Buffer.concat([from, from.slice(0, 12)]);
-      }
-      transaction.signAndHash(fakePrivateKey);
+      // if we don't have a secret key, the transaction should be signed
+      // already.
+      throw new CodedError(NO_SECRET_KEY, JsonRpcErrorCode.INVALID_INPUT);
```

Comment:
I did this originally, but it broke some tests. This is because `eth_sendRawTransaction` does not have a secret key, but rather already has a signature. So there are other valid paths where a secret key isn't necessary. However, if there's no secret key and the transaction hasn't been signed, it's now invalid and we give an error. I don't know if this code path is actually possible, though.

Reply:
Upon further discussion, we realized that if a raw transaction is sent and makes it this far into the pool, the transaction has already been constructed and verification of a proper signature (i.e. can be used to recover from address) was sent. So we should be safe to remove this check.

Reply:
:+1: